pkg/keymanagementprovider/azurekeyvault/provider.go

```diff
@@ -141,24 +141,44 @@ func (s *akvKMProvider) GetCertificates(ctx context.Context) (map[keymanagementp
 		logger.GetLogger(ctx, logOpt).Debugf("fetching secret from key vault, certName %v,  keyvault %v", keyVaultCert.Name, s.vaultURI)
 
 		// fetch the object from Key Vault
-		// GetSecret is required so we can fetch the entire cert chain. See issue https://github.com/ratify-project/ratify/issues/695 for details
-		startTime := time.Now()
-		secretBundle, err := s.kvClient.GetSecret(ctx, s.vaultURI, keyVaultCert.Name, keyVaultCert.Version)
+		certBundle, err := s.kvClient.GetCertificate(ctx, s.vaultURI, keyVaultCert.Name, keyVaultCert.Version)
 		if err != nil {
-			return nil, nil, fmt.Errorf("failed to get secret objectName:%s, objectVersion:%s, error: %w", keyVaultCert.Name, keyVaultCert.Version, err)
+			return nil, nil, fmt.Errorf("failed to get certificate objectName:%s, objectVersion:%s, error: %w", keyVaultCert.Name, keyVaultCert.Version, err)
 		}
 
-		certResult, certProperty, err := getCertsFromSecretBundle(ctx, secretBundle, keyVaultCert.Name)
-		if err != nil {
-			return nil, nil, fmt.Errorf("failed to get certificates from secret bundle:%w", err)
-		}
+		//TODO: can I replace else with a continue?
+		if !*certBundle.Attributes.Enabled {
+			fmt.Printf("debug: certificate %s version %s is disabled.", keyVaultCert.Name, keyVaultCert.Version)
+
+			isEnabled := "false"
+			startTime := time.Now()
+			lastRefreshed := startTime.Format(time.RFC3339)
+
+			metrics.ReportAKVCertificateDuration(ctx, time.Since(startTime).Milliseconds(), keyVaultCert.Name)
+			certProperty := getStatusProperty(keyVaultCert.Name, keyVaultCert.Version, isEnabled, lastRefreshed)
+			certsStatus = append(certsStatus, certProperty)
+			certMapKey := keymanagementprovider.KMPMapKey{Name: keyVaultCert.Name, Version: keyVaultCert.Version, Enabled: isEnabled}
+			certsMap[certMapKey] = []*x509.Certificate{} // empty cert chain
+		} else {
+			// GetSecret is required so we can fetch the entire cert chain. See issue https://github.com/ratify-project/ratify/issues/695 for details
+			isEnabled := "true"
+			startTime := time.Now()
+			secretBundle, err := s.kvClient.GetSecret(ctx, s.vaultURI, keyVaultCert.Name, keyVaultCert.Version)
```

Comment:
Are we making two calls now to Keyvault? One for an initial `GetCertificates` and then another for a `GetSecret`?

Reply:
That's correct. Instead of attempting to trap a specific error from `GetSecrets` I opted to use `GetCertificates` to get the `enabled` status of the certificate to determine how to add the entry to the KMP. 

To your previous comment do we want a metric for each AKV call?

Reply:
Oh ok. I see. I'm concerned with the network overhead and how that will affect upstream throttling quotas for AKV. What is the value of doing the existence check first and then calling GetSecret. Is it primarily so the disabled certificate content is not returned as part of the response from KeyVault?

Reply:
That's correct, with `GetCertificate` I can check the status of enabled without returning any of the secret data. It also avoids having to trap a particular error from AKV when the cert is disabled when `GetSecret` is called. 

That said, I'm open to suggestions. :)

Reply:
I think we can get away with a single call to `GetSecret` if we trap the 403 response and continue on instead of erroring out. The only draw back is that means we have to assume that 403 means the certificate is disabled, which is fairly safe imo. My only concern with just using `GetSecret` is if the call fails we have no information about the secret beyond what's provided in the configuration. So in the instance that a user defines a cert or key without a version we won't be able to update the status with the version of disabled certificates. 

Another option is to only make a `GetCertificate` call when a cert is disabled to gather this info. Which switches the logic from always doing 2x calls to only sometimes doing 2x calls.

Reply:
I've got this working in another [branch](https://github.com/duffney/ratify/blob/5c6c108f71099b7fb0921997d875568f0c4fddcd/pkg/keymanagementprovider/azurekeyvault/provider.go#L170), with `GetCertificate` only being called if `GetSecret` returns a 403 error indicating that the certificate is disabled.

Reply:
I feel we also need to check if the root cause is disabled secret since 403 also indicates a permission error. Not sure if this  post will help: https://stackoverflow.com/questions/71184772/handling-a-disabled-azure-key-vault-secret-using-go-azure-sdk

Reply:
@binbin-li how do you feel about `GetCertificate` being called when it's disabled to get the version info from azkv? To me, it felt like a poor user experience if the disabled certs versions are absent.